docs/recipes/reducers/00-PrerequisiteConcepts.md

```diff
@@ -0,0 +1,88 @@
+# Prerequisite Reducer Concepts
+
+
+As described in [Reducers](../../basics/Reducers.md), a Redux reducer function:
+
+- Should have a signature of `(previousState, action) => newState`, similar to the type of function you would pass to [`Array.prototype.reduce(reducer, ?initialValue)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)
+- Should be "pure", which means it does not mutate its arguments, perform side effects like API calls or modifying values outside of the function, or call non-pure functions like `Date.now()` or `Math.random()`.  This also means that updates should be done in an ***"immutable"*** fashion, which means **always returning new objects with the updated data**, rather than directly modifying the original state tree in-place.
+
+>##### Note on immutability, side effects, and mutation
+> Mutation is discouraged because it generally breaks time-travel debugging, and React Redux's `connect` function.  For time traveling, the Redux DevTools expect that replaying recorded actions would output a state value, but not change anything else.  For React Redux, `connect` checks to see if the values returned from a `mapStateToProps` function have changed in order to see if a component needs to update.  Direct mutation can cause both of these scenarios to not work correctly.  Other side effects like generating unique IDs or timestamps in a reducer also make the code unpredictable and harder to debug.  
+
+
+Because of these rules, it's important that the following core concepts are fully understood before moving on to other specific techniques for organizing Redux reducers:
+
+#### Redux Reducer Basics
+
+**Key concepts**:
+
+- Thinking in terms of state and state shape
+- Delegating update responsibility by slice of state (*reducer composition*)
+- Higher order reducers
+- Defining reducer initial state
+
+**Reading list**:
+
+- [Redux Docs: Reducers](../../basics/Reducers.md)
+- [Redux Docs: Reducing Boilerplate](../ReducingBoilerplate.md)
+- [Redux Docs: Implementing Undo History](../ImplementingUndoHistory.md)
+- [Redux Docs: `combineReducers`](../../api/combineReducers.md)
+- [The Power of Higher-Order Reducers](http://slides.com/omnidan/hor#/)
+- [Stack Overflow: Store initial state and `combineReducers`](http://stackoverflow.com/questions/33749759/read-stores-initial-state-in-redux-reducer)
+- [Stack Overflow: State key names and `combineReducers`](http://stackoverflow.com/questions/35667775/state-in-redux-react-app-has-a-property-with-the-name-of-the-reducer)
+
+
+#### Pure Functions and Side Effects
+
+**Key Concepts**:  
+
+- What "side effects" are
+- What "pure functions" are
+- How to think in terms of combining functions
```

Comment:
IMO this reads slightly better as:

**Key Concepts**:  

- Side effects
- Pure functions
- How to think in terms of combining functions